browser/ipfs/ipfs_service_factory.cc

```diff
@@ -5,20 +5,63 @@
 
 #include "brave/browser/ipfs/ipfs_service_factory.h"
 
+#include "base/feature_list.h"
 #include "base/path_service.h"
 #include "brave/browser/brave_browser_process_impl.h"
 #include "brave/browser/profiles/profile_util.h"
+#include "brave/components/ipfs/features.h"
+#include "brave/components/ipfs/ipfs_constants.h"
 #include "brave/components/ipfs/ipfs_service.h"
+#include "brave/components/ipfs/pref_names.h"
 #include "chrome/browser/profiles/incognito_helpers.h"
 #include "chrome/common/chrome_paths.h"
 #include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
 #include "extensions/browser/extension_registry_factory.h"
 #include "extensions/browser/extension_system.h"
 #include "extensions/browser/extension_system_provider.h"
 #include "extensions/browser/extensions_browser_client.h"
 
 namespace ipfs {
 
+// static
+bool IpfsServiceFactory::IsIpfsDisabledByPolicy() {
+  if (!g_brave_browser_process)
+    return false;
+
+  PrefService* local_state = g_brave_browser_process->local_state();
+
+  // Because we currently do not provide a settings switch for IPFSEnabled
+  // preference to be overwritten by users, the policy is configured that only
+  // a mandatory value can be set by admins.
+  return local_state->IsManagedPreference(kIPFSEnabled) &&
```

Comment:
I didn't expose a new settings switch cuz it will be confusing.
Users are still able to set the resolve method (per-profile setting) as disabled in the settings which will be respected immediately without restarting the browser.
This new kIPFSEnabled pref is only for admins to add a mandatory policy to disable the whole IPFS feature for the machine and will remove the binary if already downloaded.